common/src/main/java/com/emc/mongoose/common/concurrent/DaemonBase.java

```diff
@@ -12,6 +12,7 @@
 implements Daemon {
 
 	private AtomicReference<State> stateRef = new AtomicReference<>(State.INITIAL);
+	protected final Object state = new Object();
 
 	private enum State {
 		INITIAL, STARTED, SHUTDOWN, INTERRUPTED, CLOSED
```

Comment:
this is just the subject to lock on only - fake object
this is a common practice to have an Object field to lock/signal/wait for any state changes
AFAIK it's not recommended to synchronize on "this"